wallet/wallet.go

```diff
@@ -3224,6 +3224,13 @@ func (w *Wallet) PublishTransaction(tx *wire.MsgTx) error {
 	_, err = server.SendRawTransaction(tx, false)
 	switch {
 	case err == nil:
+		switch w.chainClient.(type) {
+		// For neutrino we need to trigger adding relevant tx manually
+		// because for spv client - tx data isn't received from sync peer.
+		case *chain.NeutrinoClient:
+			return w.AddRelevantTx(txRec, nil)
```

Comment:
Why's a new method needed here? Why not just call the existing one as is, wrapped in a DB transaction?

Reply:
Looking at this method again, it seems to be the case that above, we should actually be calling `addRelevantTx` as well. This ensure that we'll always notify the clients, and also that addresses will be marked as used, and credits added.

Reply:
I believe `addRelevantTx` is called once the `chain.RelevantTx` notification is received, which is sent once we filter it from the backend.

Reply:
@Roasbeef I've removed method

yep `addRelevantTx` is called when notification is received, for `btcd` - `onRecvTx` for `bitcoind` - `onRelevantTx`